public/app/features/explore/RawListContainer.tsx

```diff
@@ -0,0 +1,116 @@
+  tableResult: DataFrame;
+}
+
+type instantQueryMetricList = { [index: string]: { [index: string]: instantQueryRawVirtualizedListData } };
+
+const getRawListContainerStyles = stylesFactory(() => {
+  return {
+    wrapper: css`
+      height: 100%;
+      overflow: scroll;
+    `,
+  };
+});
+
+/**
+ * transform dataFrame to instantQueryRawVirtualizedListData
+ * @param dataFrame
+ */
+const getListItemsFromDataFrameNew = (dataFrame: DataFrame): instantQueryRawVirtualizedListData[] => {
+  const metricList: instantQueryMetricList = {};
+  const outputList: instantQueryRawVirtualizedListData[] = [];
+
+  // Filter out time
+  const newFields = dataFrame.fields.filter((field) => !['Time'].includes(field.name));
+
+  // Get name from each series
+  // const metricNames: string[] = newFields.find((field) => ['__name__'].includes(field.name))?.values?.toArray() ?? [''];
+  let metricNames: string[] = newFields.find((field) => field.name === '__name__')?.values.toArray() ?? [];
+  // const metricNames: string[] = doesMetricHaveNames ?
+  if (!metricNames.length) {
+    // These results do not have series labels
+    // Matching the native prometheus UI which appears to only show the permutations of the first field in the query result.
+    metricNames = Array(newFields[0].values.length).fill('');
+  }
+
+  // Get everything that isn't the name from each series
+  const metricLabels = dataFrame.fields.filter((field) => !['__name__'].includes(field.name));
+
+  metricNames.forEach(function (metric: string, i: number) {
+    metricList[metric] = {};
+    const formattedMetric: instantQueryRawVirtualizedListData = metricList[metric][i] ?? {};
+
+    for (const field of metricLabels) {
+      const label = field.name;
+
+      if (label !== 'Time') {
+        // Initialize the objects
+        if (typeof field?.display === 'function') {
+          const stringValue = formattedValueToString(field?.display(field.values.get(i)));
+          if (stringValue) {
+            formattedMetric[label] = stringValue;
+          }
+        } else {
+          console.warn('Field display method is missing!');
+        }
+      }
+    }
+
+    outputList.push({
+      ...formattedMetric,
+      __name__: metric,
+    });
+  });
+
+  return outputList;
+};
+
+/**
+ * The container that provides the virtualized list to the child components
+ * @param props
+ * @constructor
+ */
+const RawListContainer = (props: RawListContainerProps) => {
+  const { tableResult } = props;
+  let dataFrame = cloneDeep(tableResult);
+  const styles = getRawListContainerStyles();
+  const items = getListItemsFromDataFrameNew(dataFrame);
+
+  return (
+    // We don't use testids around here, how should we target this element in tests?
+    <section data-testid={'raw-list-container'}>
+      <div>Result series: {items.length}</div>
+
+      <List
+        itemCount={items.length}
+        className={styles.wrapper}
+        itemSize={(index: number) => {
+          return 42;
```

Comment:
[The answer for everything](https://simple.wikipedia.org/wiki/42_(answer)) 😂 Jokes apart. Do you know if we have some theming constant for this case? Or `42` is needed to go? If so, you can use just `itemSize={(index: number) => 42}` as read improvement.

Reply:
Yeah 42 worked out to be a good number so I went with it, it's a good amount of spacing above and below the text, and gives the copy icon enough room as well. Should be updated now